config.go

```diff
@@ -662,11 +658,10 @@ func (s *DBSettings) mongoOptions(url string) *options.ClientOptions {
 		SetReadConcern(s.ReadConcernSettings.Resolve()).
 		SetTimeout(mongoTimeout).
 		SetConnectTimeout(mongoConnectTimeout).
-		// SetSocketTimeout will be deprecated in future Go driver releases, though at the time being there
-		// isn't any other way to enforce a time limit on how long the client waits when trying to R/W data
-		// over a connection, so we are including it until Go driver finalizes their timeout API.
-		SetSocketTimeout(mongoTimeout).
-		SetMonitor(apm.NewMonitor(apm.WithCommandAttributeDisabled(false), apm.WithCommandAttributeTransformer(redactSensitiveCollections)))
+		SetMonitor(apm.NewMonitor(apm.WithCommandAttributeDisabled(false), apm.WithCommandAttributeTransformer(redactSensitiveCollections))).
+		SetBSONOptions(&options.BSONOptions{
+			ObjectIDAsHexString: true,
+		})
```

Comment:
SetSocketTimeout is removed now, with SetTimeout, SetConnectTimeout, and using context's with max time replace it's functionality.

Reply:
As well, by default the encoder/decoder no longer treat ObjectIDs as hex strings and you need to set this option. This isn't saying all object ids as hex strings, but allow the compatability for it.

Reply:
Is the implication here that if it's stored in BSON as an ObjectId, but our data model expects a string, it'll decode the ObjectId to its hex representation automatically? Is there any place that relies on this currently?

Reply:
I can't comment on the comment you made, so I'm doing here:
> Is the implication here that if it's stored in BSON as an ObjectId, but our data model expects a string, it'll decode the ObjectId to its hex representation automatically? Is there any place that relies on this currently?

Yes. Basically all the usages of mgobson (our internal driver) that use a string as id requires this to be set. I originally didn't have the option set, and it makes quite a bit of tests fail ([this](https://spruce.mongodb.com/version/67c871668c2db0000751b1b3/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&variant=%5Eubuntu2204%24) was before, most failures from the decoder/encoder not handling id fields being strings)